client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -121,38 +172,47 @@ class InlineHelpSearchResults extends Component {
 		return selectedLink.link;
 	}
 
-	componentDidMount() {
-		this.props.setSearchResults( '', this.getContextResults() );
-	}
-
-	componentWillUpdate( nextProps ) {
-		if ( nextProps.shouldOpenSelectedResult ) {
-			const url = this.getSelectedUrl();
-			this.followHelpLink( url )();
-			this.props.didOpenResult();
-			window.location = url;
-			return;
-		}
-	}
-
-	followHelpLink = ( url ) => {
-		const payload = {
-			searchQuery: this.props.searchQuery,
-			currentUrl: window.location.href,
-			resultUrl: url,
-		};
+	followHelpLink = ( url, index ) => {
+		const railcar = this.getRailcarForIndex( index );
 		return () => {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', payload );
+			// TrainTracks interact event
+			this.props.recordTracksEvent( 'calypso_traintracks_interact', {
+				action: 'calypso_inlinehelp_link_open',
+				railcar: railcar,
+			} );
+			// regular Tracks event
+			this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', {
+				search_query: this.props.searchQuery,
+				current_url: window.location.href,
+				result_url: url,
+			} );
 		};
 	}
 
 	renderHelpLink = ( link, index ) => {
+		// TrainTracks render event
+		const railcar = this.getRailcarForIndex( index );
+		const isShowingSuggestions = this.isShowingSuggestions();
+		const event = () => {
+			this.props.recordTracksEvent( 'calypso_traintracks_render', {
```

Comment:
Hmmm, did I get confused, are you not getting results from an API? 

The railcar and all of the properties should be getting generated by the API so we can change the API fetch_algo without needing to change the client.